src/persist/src/file.rs

```diff
@@ -152,10 +156,18 @@ impl FileBuffer {
     fn metadata_path(base_dir: &Path) -> PathBuf {
         base_dir.join(Self::METADATA_PATH)
     }
+
+    fn ensure_open(&self) -> Result<PathBuf, Error> {
+        self.base_dir
+            .clone()
+            .ok_or_else(|| return Error::from("FileBuffer unexpectedly closed"))
+    }
 }
 
 impl Buffer for FileBuffer {
     fn write_sync(&mut self, buf: Vec<u8>) -> Result<SeqNo, Error> {
+        let _ = self.ensure_open()?;
```

Comment:
I think this can just be `self.ensure_open()?;`

Reply:
Done